osbs/build/user_params.py

```diff
@@ -97,18 +111,90 @@ def value(self, val):  # pylint: disable=W0221
         BuildParam.value.fset(self, registry_uris)  # pylint: disable=no-member
 
 
+def load_user_params_from_json(user_params_json):
+    """Load user params from json into proper object
+
+    :param str user_params_json: json with user params
+    :rtype: subclass of BuildCommon
+    :return: initialized object with user params
+    """
+    json_dict = json.loads(user_params_json)
+    kind = json_dict.get(KIND_KEY, BuildUserParams.KIND)  # BW comp. default to BuildUserParams
+    user_params_class = user_param_kinds[kind]
+    user_params = user_params_class()
+    user_params.from_json(user_params_json)
+    return user_params
+
+
 class BuildCommon(object):
-    def __init__(self):
+
+    @abstractproperty
+    def KIND(self):
+        return 'DEFINE_KIND_NAME_IN_SUBCLASS'
+
+    def __init__(self, build_json_dir=None):
+        self.arrangement_version = BuildParam(
+            "arrangement_version",
+            allow_none=True,
+            default=REACTOR_CONFIG_ARRANGEMENT_VERSION)
+        self.build_json_dir = BuildParam('build_json_dir', default=build_json_dir)
+        self.kind = BuildParam(KIND_KEY, default=self.KIND)
+        self.component = BuildParam('component')
+        self.filesystem_koji_task_id = BuildParam("filesystem_koji_task_id", allow_none=True)
         self.image_tag = BuildParam("image_tag")
         self.koji_target = BuildParam("koji_target", allow_none=True)
+        self.koji_task_id = BuildParam('koji_task_id', allow_none=True)
         self.platform = BuildParam("platform", allow_none=True)
-        self.arrangement_version = BuildParam("arrangement_version", allow_none=True)
-        self.filesystem_koji_task_id = BuildParam("filesystem_koji_task_id", allow_none=True)
+        self.orchestrator_deadline = BuildParam('orchestrator_deadline', allow_none=True)
+        self.scratch = BuildParam('scratch', allow_none=True)
         self.user = UserParam()
-        self.component = BuildParam('component')
+        self.worker_deadline = BuildParam('worker_deadline', allow_none=True)
+
         self.required_params = [
+            self.build_json_dir,
             self.koji_target,
+            self.user,
         ]
+        self.convert_dict = {}
+
+    def attrs_finalizer(self):
+        for _, param in self.__dict__.items():
+            if isinstance(param, BuildParam):
+                # check that every parameter has a unique name
+                if param.name in self.convert_dict:
+                    raise OsbsValidationException(
```

Comment:
minor: this is not tested (was not tested before).

Reply:
> I am wondering why the TODO for "atomic-reactor user_params.json needs update or create a new jsonschema for source containers" has a strike through now

Because it requires more work than I originally expected, so it shouldn't block this PR. Will be addressed separately